Tests/Common/IsolatorLimitResultProviderTests.cs

```diff
@@ -122,15 +122,18 @@ public void ConsumesMultipleMinutes()
         {
             var timeProvider = new ManualTimeProvider(new DateTime(2000, 01, 01));
             var provider = new FakeIsolatorLimitResultProvider();
+            var timeMonitorEvent = new AutoResetEvent(false);
+            _timeMonitor.TimeMonitorEvent = timeMonitorEvent;
+
             Action code = () =>
             {
                 // lets give the monitor time to register the initial time
-                Thread.Sleep(100);
+                timeMonitorEvent.WaitOne();
                 for (int i = 0; i < 4; i++)
                 {
                     timeProvider.AdvanceSeconds(45);
                     // give the monitoring task time to request more time
-                    Thread.Sleep(20);
+                    timeMonitorEvent.WaitOne();
```

Comment:
Awesome to see these sleeps removed! ✨ 
The improvements could probably be expanded into all the tests in this fixture?